### PR TITLE
Updated Upstream (Bukkit/CraftBukkit)

### PR DESCRIPTION
**Pull request by Machine-Maker**
Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
19e985ec SPIGOT-6746: Clarify PlayerPortalEvent documentation
2a0a2e2e SPIGOT-6738: Add entity type tags

CraftBukkit Changes:
dc764e7a PR-877: Improve and simplify CraftBlockState
2d933bae SPIGOT-6741: shouldGenerateStructures isn't implemented
9aeb46ae SPIGOT-6023, SPIGOT-6745: Fix missing BlockDispenseArmorEvents
d3cc4120 SPIGOT-6738: Add entity type tags

### CODE REVIEW

**Review by kennytv — APPROVED** [review]

**Comment by Machine-Maker** [comment]
Needs to be re-reviewed with new upstream changes

File: patches/server/0168-API-to-get-a-BlockState-without-a-snapshot.patch [hunk 1]
```diff
@@ -58,76 +55,57 @@ index 77645019c88d61dde28b7598d8a29b7d0c23c209..8a079ee3ed243fd19b1dd7eed2de1dd3
...
+@@ -10,15 +10,26 @@ public abstract class CraftBlockEntityState<T extends BlockEntity> extends Craft
  
+     private final T tileEntity;
+     private final T snapshot;
 +    public final boolean snapshotDisabled; // Paper
-+    public static boolean DISABLE_SNAPSHOT = false; // Paper
-+
-     public CraftBlockEntityState(Material material, T tileEntity) {
-         super(material);
++    public static boolean DISABLE_SNAPSHOT = false; // paper
```
Comment (kennytv) [inline patches/server/0168-API-to-get-a-BlockState-without-a-snapshot.patch]: Make the p uppercase again

File: patches/server/0606-Added-Vanilla-Entity-Tags.patch [hunk 2]
```diff
@@ -38,14 +39,31 @@ index 0000000000000000000000000000000000000000..6271586368c65250c887739d04c5fccf
...
 +}
+diff --git a/src/main/java/net/minecraft/server/ServerScoreboard.java b/src/main/java/net/minecraft/server/ServerScoreboard.java
+index 130a928f156961bae9ca184b3ca31004dbba1012..da1bd59df32e94180c2c19aa64b4338f57316c09 100644
+--- a/src/main/java/net/minecraft/server/ServerScoreboard.java
++++ b/src/main/java/net/minecraft/server/ServerScoreboard.java
+@@ -40,6 +40,12 @@ public class ServerScoreboard extends Scoreboard {
+         this.setDirty();
+     }
+ 
++    public void test(java.util.Collection<String> entries, PlayerTeam team) {
```
Comment (kennytv) [inline patches/server/0606-Added-Vanilla-Entity-Tags.patch]: 🦧

**Review by kennytv — APPROVED** [review]
Should be fine other than those two

**Review by MiniDigger — APPROVED** [review]

**Pull request merged**